pennylane_lightning/lightning_qubit/lightning_qubit.py

```diff
@@ -117,27 +117,6 @@
         "CRX",
         "CRY",
         "CRZ",
-        "C(PauliX)",
```

Comment:
(This is more a question, I am not asking for immediate changes in this PR)

The set of supported gates (in Lighthing at least) is defined [here](https://github.com/PennyLaneAI/pennylane-lightning/blob/00aaa927d73d6eab6813c3cb12ad1755ac7641fa/pennylane_lightning/core/src/gates/GateOperation.hpp#L27) and in C++ (and I would say that the complete spec would also include threading model and the number of qubits). Is the Python list below - a duplicate of these C++ enums? 
Could we use the device API to read the supported gates somehow? Maybe, we can autogenerate Python sources from the C++ ones if we want to have this information in compile time without running C funcitons?

Reply:
The problem is we do not have a tight and robust mechanism to communicate natively supported gates. The `enum`s are just that, `enum`s. So they are not lists of the supported gates, just gates that might be supported on some device. What dictates which gates are decomposed are the `allowed_operations` dictionaries, but these are manually maintained and might not include gates that are natively supported (usually goes uncaught), and vice versa (usually fails some test). Some gates like `OrbitalRotation` are listed in `allowed_operations` but are not natively supported and do not crash any test. This is because gates that do not have a specialized implementation are applied via matrices. In the case of `OrbitalRotation`, this is fine because the number of wire is fixed at 4, a reasonable number. For other gates however, like `QFT`, this can cause issues since the number of wire is unlimited, possibly requiring the generation of a large dense matrix at the Python layer (filling out the memory).